### ISSUE TITLE
augmentation.RandomMotionBlur: reflect and circular borders raise raw padding errors on small images

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4784]
The #4559 fix gave `RandomBoxBlur`, `RandomGaussianBlur` and `RandomSharpness` a named `ValueError` when the image is too small for the kernel under the chosen border type. `RandomMotionBlur` (and `kornia.filters.motion_blur` under it) has the same precondition under `border_type="reflect"` and `"circular"` but still surfaces torch's raw padding errors. The #4559 thread recorded the motion-blur cases, but the issue closed with only the three classes fixed.

## Repro

```python
import torch, kornia, kornia.augmentation as K

torch.manual_seed(0)
for shape, border, k in [((2, 3, 1, 8), "reflect", 3), ((2, 3, 1, 1), "circular", 5), ((2, 3, 1, 8), "constant", 3)]:
    try:
        K.RandomMotionBlur(k, 35.0, 0.5, border_type=border, p=1.0)(torch.rand(*shape))
        print(shape, border, k, "ok")
    except Exception as e:
        print(shape, border, k, type(e).__name__, e)
try:
    K.RandomBoxBlur(p=1.0)(torch.rand(2, 3, 1, 8))
except Exception as e:
    print("RandomBoxBlur", type(e).__name__, e)
```

## Observed

```
(2, 3, 1, 8) reflect 3 RuntimeError Argument #6: Padding size should be less than the corresponding input dimension, but got: padding (1, 1) at dimension 2 of input [2, 3, 1, 8]
(2, 3, 1, 1) circular 5 RuntimeError Padding value causes wrapping around more than once.
(2, 3, 1, 8) constant 3 ok
RandomBoxBlur ValueError RandomBoxBlur cannot filter an image this small: kernel_size=(3, 3) with border_type='reflect' needs at least 2 pixel(s) along height, but the input is (2, 3, 1, 8).
```

## Expected

The same named check the sibling blurs got in the #4559 fix: `reflect` needs each spatial axis longer than the kernel radius, `circular` at least the radius; `constant` (the default) and `replicate` need nothing.

## Environment

kornia `main` @ 57cd68e6f, torch 2.14.0, python 3.11, macOS arm64 cpu.

Posted on behalf of @ducha-aiki by Claude (Opus 5.5).
